### DISCUSSION TITLE
Detect current IM via `fcitx5-remote`?

### DISCUSSION THREAD

**Discussion by fosskers** [fcitx/fcitx5#624] [General]
Hi there, thank you for this great project.

I see that we're able to detect the current IM _group_ on the command line via:
```
> fcitx5-remote -q
Default
```
but is there a way to print a string of the current active IM?

**Comment by wengxt** [comment]
Thanks, I just added the support in https://github.com/fcitx/fcitx5/commit/c1b3b8f90579e1a6197e9f6a9454dab45f2af7fa

Meanwhile, you can query dbus for this information: 
e.g.
`qdbus org.fcitx.Fcitx5 /controller org.fcitx.Fcitx.Controller1.CurrentInputMethod`
(dbus-send or busctl can also be used.)

**Reply by fosskers** [reply]
Thank you very much!

**Reply by fosskers** [reply]
I couldn't get it to work via `dbus-send`, but I built `master` and it's working great. Thank you!

**Reply by wengxt** [reply]
`dbus-send --session --print-reply --dest=org.fcitx.Fcitx5 /controller org.fcitx.Fcitx.Controller1.CurrentInputMethod`

`busctl --user   call org.fcitx.Fcitx5 /controller org.fcitx.Fcitx.Controller1 CurrentInputMethod`

Command line  is somewhat different and they have different output format though. (That's why I prefer qdbus )
